### ISSUE TITLE
Consider to support change of maven target directory

### ISSUE THREAD

**Issue by mikir** [xerial/sqlite-jdbc#825]
I would like to change default maven output directory `target` to something else. For example by using profiles:

```
<profiles>
    <profile>
        <id>otherOutputDir</id>
        <build>
            <directory>myDirectory</directory>
        </build>
    </profile>
</profiles>
```
If I do this, JDBC tests fail. This is because JDBC tests are hardcoded to use `target` output directory. For example:

```
[ERROR] Tests run: 4, Failures: 0, Errors: 2, Skipped: 2, Time elapsed: 0.065 s <<< FAILURE! - in org.sqlite.BusyHandlerTest
[ERROR] testRemovingBusyHandler  Time elapsed: 0.052 s  <<< ERROR!
java.sql.SQLException: path to 'target/test0.db': '/home/Projects/SQLiteJdbc/target' does not exist
	at org.sqlite.BusyHandlerTest.createConnection(BusyHandlerTest.java:39)
	at org.sqlite.BusyHandlerTest.connect(BusyHandlerTest.java:27)
```

It would be helpfull if JDBC tests can be improved to pass even if no `target` directory exists.

**Comment by gotson** [comment]
> I would like to change default maven output directory `target` to something else

why would you do that?

**Comment by gotson** [comment]
> My use case is that I have two different proprietary SQLite distributions (for the same platform) for which I would like to compile corresponding JDBC driver.

i don't quite understand. That would be 2 different projects, in 2 different directory. Or did i miss something?

**Comment by mikir** [comment]
Sorry for unclear explanation.

I have two different SQLite native libraries for one platform. Let's say `libsqlitejdbcA.so` and `libsqlitejdbcB.so`. I would like to compile the same JDBC driver sources twice to get `sqlite-jdbc-A.jar` and `sqlite-jdbc-B.jar`. My current solution is:

- I compile JDBC driver with `libsqlitejdbcA.so`
- I rename the output directory `target` to `targetA`
- I compile JDBC driver with `libsqlitejdbcB.so`
- I rename the output directory `target` to `targetB`

Everything should be done automatically in some script to be able to use it in CI. So in another words, I would like to have two different output directories for one JDBC project.

**Comment by gotson** [comment]
i still don't understand what you are doing, but hey, we do accept PRs :)

**Comment by mikir** [comment]
Thanks! I know I have a peculiar use case. :-)

Maybe the important thing to mention is that I don't have public SQLite. I have proprietary SQLite for which I would like to compile JDBC. And unfortunatelly, I have two completelly different proprietary SQLite libraries per each platform for which I need to compile JDBC.

So, I would like to inject my SQLite native libraries and not to use native libraries from `src/main/resources`. One can adapt `pom.xml` to fullfill this:

```
    <properties>
        <project.build.nativeLibDirectory>${project.basedir}/src/main/resources</project.build.nativeLibDirectory>
    </properties>

    <resource>
        <directory>${project.build.nativeLibDirectory}</directory>
        <includes>
            <include>org/**</include>
        </includes>
    </resource>
```

Then, I can easily compile JDBC for two different sets of SQLite native libraries by the following commands:

```
mvn package -Dproject.build.nativeLibDirectory=<MY_NATIVE_LIB_A_DIR>
mvn package -Dproject.build.nativeLibDirectory=<MY_NATIVE_LIB_B_DIR>
```

Therefore, it would be handy to be able to change maven output directory not to overwrite `target` directory with the second run of `mvn`.

**Comment by gotson** [comment]
> I have proprietary SQLite for which I would like to compile JDBC

You only need to compile the native lib, not the Java project.

You can load any custom native lib using https://github.com/xerial/sqlite-jdbc/blob/master/USAGE.md#how-to-use-a-specific-native-library at runtime.

**Comment by gotson** [comment]
Regarding the original ticket scope, not to depend on `target`, there's a couple of complications:
- ArchUnit defines [those](https://github.com/TNG/ArchUnit/blob/6f32fbd8bad4027abcef100322cc32fcfed2c25b/archunit/src/main/java/com/tngtech/archunit/core/importer/ImportOption.java#L101-L105) which won't work when you change the build directory.
- our `MultipleClassLoaderTest` needs to [find the classes under test](https://github.com/xerial/sqlite-jdbc/blob/b13c908f0921efa75eed723badaa3ead6f549e93/src/test/java/org/sqlite/MultipleClassLoaderTest.java#L75-L75), and does so by checking the `target` folder.

**Comment by mikir** [comment]
> You can load any custom native lib using https://github.com/xerial/sqlite-jdbc/blob/master/USAGE.md#how-to-use-a-specific-native-library at runtime.

Oh, thanks a lot for a hint! I will think about it.

Although I forgot about this possibility, my `libsqlitejdbc.so` depends on others `.so` libraries. Therefore, I had to change a little bit `SQLiteJDBCLoader` to extract all libraries available in `jar` file.

**Comment by gotson** [comment]
> > You can load any custom native lib using https://github.com/xerial/sqlite-jdbc/blob/master/USAGE.md#how-to-use-a-specific-native-library at runtime.
> 
> Oh, thanks a lot for a hint! I will think about it.
> 
> Although I forgot about this possibility, my `libsqlitejdbc.so` depends on others `.so` libraries. Therefore, I had to change a little bit `SQLiteJDBCLoader` to extract all libraries available in `jar` file.

You would probably be better using this flag, and distribute the necessary so files separately.

**Comment by gotson** [comment]
I have reverted the changes as the CI broke on Windows because of https://github.com/junit-team/junit5/issues/2811

**Comment by gotson** [comment]
Fixed by #839
